runners/direct-java/src/main/java/org/apache/beam/runners/direct/WindowEvaluatorFactory.java

```diff
@@ -52,7 +54,18 @@
   private <InputT> TransformEvaluator<InputT> createTransformEvaluator(
       AppliedPTransform<PCollection<InputT>, PCollection<InputT>, Window.Assign<InputT>>
           transform) {
-    WindowFn<? super InputT, ?> fn = transform.getTransform().getWindowFn();
+
+    WindowFn<? super InputT, ?> fn;
+    try {
+      fn = (WindowFn) WindowIntoTranslation.getWindowFn(transform);
+    } catch (IOException exc) {
+      throw new IllegalArgumentException(
+          String.format(
+              "%s received transform that did not contain a Java-serialized %s: %s",
```

Comment:
That specific reason seems like an exception that should be thrown out of the `getWindowFn` call, as it's returning an object of type `WindowFn`; This should just report that it couldn't get the WindowFn out of the transform and attach the cause.